python/ray/serve/backend_state.py

```diff
@@ -156,6 +155,14 @@ def check_stopped(self) -> bool:
 
         return self._stopped
 
+    def check_health(self) -> bool:
+        """Check if the actor is healthy."""
+        if self._health_check_ref is None:
+            self._health_check_ref = self._actor_handle.run_forever.remote()
+
+        ready, _ = ray.wait([self._health_check_ref], timeout=0)
+        return len(ready) == 0
```

Comment:
i'm not sure about this part. We are relying on that when the worker fail, Ray will reliably set the `_health_check_ref` to error, which relies on the fault tolerance capability of Ray. 

Should we just perform periodic health check and actually expect the actors to return OK:
```
ref = self._actor_handle._health_check.remote()
...
await ref
```
This way, we are relying on the regular task submission code path instead of Ray's fault tolerance functionality.

Reply:
I don't see a problem with relying on Ray's failure detection logic. We're already relying on this for things like node failures and restarting the controller. The existing issue is an issue of semantics, not the correctness of the mechanism.

Reply:
Doing our own heartbeats would be significantly more complex and would also have worse performance (we'd need to configure some kind of heartbeat timeout).